### PR TITLE
Fix incorrect data node rendering for modular pipelines 

### LINKED ISSUE #1123: Investigate incorrect rendering of dataset node seen on a project with nested modular pipelines
## Description
After applying the fix (#1119) for [#1036](https://github.com/kedro-org/kedro-viz/issues/1036), another minor error in rendering is seen in the project used to replicate the bug.

On initial loading of the project, the data node 'Preprocessed Data' is visible, although it should be included in the 

Screenshot 2022-10-05 at 19 44 12

After expanding the 'Sub Pipeline' modular pipeline once, the data node is rendered correctly once it is collapsed.
Screenshot 2022-10-05 at 20 10 14

I discussed this with @tynandebold, though we couldn't conclude if this is due to a bug in the tree generation code on the backend, or a bug in the rendering code on the frontend.

It could be an error on the backend because the generated tree includes the errant node as a child of both the default pipeline (`__root__`) and in the modular pipeline `sub_pipeline`. 

It could be an error on the frontend because the node is rendered correctly after one expansion / collapse of the modular pipeline.

This should be discussed in a tech design session.

In the case that it is an error on the backend, completing ticket #1122 will be a helpful first step in diagnosing the cause.
 
## Context
The bug is minor and has only been seen in this case, but could have deeper implications for the correctness of the code that causes it.

## Steps to Reproduce

1. Create a local project with the code given in #1119.
2. Run Viz.
3. Expand and collapse the 'Sub Pipeline' modular pipeline.

## Expected Result
The pipeline shown in the second image above should render first time.

## Actual Result
The pipeline shown in the first image above is rendered.

## Your Environment
Include as many relevant details as possible about the environment you experienced the bug in:

* Web browser system and version: 103.0.5060.114
* Operating system and version:  macOS 12.6
* NodeJS version used (if relevant):  16.17.0
* Kedro version used (if relevant): 0.18.3
* Python version used (if relevant): 3.7

## Checklist
- [x] Include labels so that we can categorise your issue

### PR DESCRIPTION
**Pull request by rashidakanchwala**
## Description

Fixes #1123 

## Development notes

The viz flowchart is actually a representation of modular pipelines. The `root` modular pipeline is the first thing we see in the collapsed flowchart. It must include:

- All top-level modular pipelines (e.g., level1.node, not level2.level1.node).
- Any tasks, datasets, or parameters not part of a modular pipeline.
- Inputs and outputs to the top-level modular pipeline (also called `external_inputs` and `external_outputs` in the code).
- Lastly, all `parameters`, even if they belong to nested modular pipelines. The flowchart design ensures these parameters remain visible from the `root` modular pipeline, even when we expand the view.

In the code, there was a variable named `root_children_ids`, which was causing issues as reported by users. It's a bit confusing what it does but removing it resulted in the `parameters` not being shown at the `root` level anymore.

To address this, the variable `root_children_ids` has been renamed to `root_parameters`, and it has been modified to include all parameters that are part of a modular pipeline. 

After making this change, the code correctly satisfies the above four conditions, ensuring all relevant components are present in the root modular pipeline.

The dummy project the original user posted in the original issue available on box. Please ping me for the link to it

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1439"></a>

### CODE REVIEW

File: package/kedro_viz/data_access/managers.py [hunk 1]
```diff
@@ -456,7 +456,9 @@ def create_modular_pipelines_tree_for_registered_pipeline(
                 or not node.belongs_to_pipeline(registered_pipeline_id)
             ):
                 continue
-            if not node.modular_pipelines or node_id in root_children_ids:
```
Comment (rashidakanchwala) [inline package/kedro_viz/data_access/managers.py:459]: The root modular pipeline is what we first see in the collapsed Kedro-viz pipeline. Initially, users should be able to view:

- Top-level modular pipelines
- Any tasks, datasets, or parameters that are not part of a modular pipeline
- Inputs and outputs of the top-level modular pipeline

The above if statement works as follows:
- It adds all nodes that don't belong to a modular pipeline to root
- It also adds inputs and outputs of a modular pipeline, including inputs and outputs of nested modular pipelines
-
The main distinction between inputs/outputs of top-level modular pipelines and inputs/outputs of nested modular pipelines is that the former doesn't have any associated modular pipeline. 

Therefore, we need to revise this if statement to include:
- Nodes that are not part of any modular pipeline
- Inputs and outputs of a modular pipeline (root_children_ids) that are also not part of the modular pipeline.

File: package/tests/test_api/test_rest/test_responses.py [hunk 2]
```diff
@@ -200,6 +200,7 @@ def assert_example_data(response_data):
             "children": [
                 {"id": "0ecea0de", "type": "data"},
                 {"id": "f1f1425b", "type": "parameters"},
+                {"id": "f0ebef01", "type": "parameters"},
```
Comment (merelcht) [inline package/tests/test_api/test_rest/test_responses.py:203]: So was part of the problem that the `expected_modular_pipelines` in this test wasn't actually correct?

**Review by merelcht — APPROVED** [review]
The code looks good to me 👍 Maybe add this to the release notes?

**Review by ravi-kumar-pilla — APPROVED** [review]

File: RELEASE.md [hunk 3]
```diff
@@ -6,6 +6,11 @@ Please follow the established format:
 - Include the ID number for the related PR (or PRs) in parentheses
 -->
 
+
+## Bug fixes and other changes
+
+- Fix the incorrect rendering of datasets in the modular pipeline. (#1439)
```
Comment (tynandebold) [inline RELEASE.md:12]: Nit.

```suggestion
- Fix incorrect rendering of datasets in modular pipelines. (#1439) 
```

**Review by tynandebold — APPROVED** [review]
Code looks good.

I couldn't actually test the UI locally with the `basic_spaceflights` project you provided, as I keep getting this warning even after installing your local branch of the package.

```
WARNING: You are using an old version of Kedro Viz. You are using version 4.7.1; however, version 6.3.4 is now available.
```

If you can provide further clarity on how to test that would be great.

**Pull request merged**